src/components/LinkButton/LinkButton.tsx

```diff
@@ -0,0 +1,87 @@
+import { ButtonProps } from '@dhis2/ui'
+import cx from 'classnames'
+import React, { AnchorHTMLAttributes } from 'react'
+import { useLinkClickHandler, useHref } from 'react-router-dom'
+import css from './LinkButton.module.css'
+
+type UseLinkClickHandlerParameters = Parameters<typeof useLinkClickHandler>
+
+type LinkClickHandlerOptions = UseLinkClickHandlerParameters[1]
+
+type RelevantButtonProps = Pick<
+    ButtonProps,
+    | 'disabled'
+    | 'className'
+    | 'primary'
+    | 'secondary'
+    | 'small'
+    | 'toggled'
+    | 'large'
+    | 'destructive'
+>
+
+type LinkButtonProps = AnchorHTMLAttributes<HTMLAnchorElement> &
+    LinkClickHandlerOptions &
+    RelevantButtonProps & {
+        to: Parameters<typeof useLinkClickHandler>[0]
+    }
+
+/* Wrapping button with anchor-tags are not valid, style anchor as a UI-button */
+export const LinkButton = ({
```

Comment:
This should be done in UI, eg passing an `as={'a'}` or something similar to button component...